### PR TITLE
feat(language-server): support Central Config API [IDE-1639]

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

Complete the migration of configuration communication to Language Server Protocol Version 25. This PR closes all 6 remaining gaps in the protocol v25 configuration migration (IDE-1639).

#### What changed

**Production code (1 fix):**
- Fix missing `LanguageServerWrapper` import in `SaveConfigHandler.kt` that caused compilation failure

**Test coverage (+2,228 lines across 6 test files):**

| Gap | Description | Tests Added |
|-----|-------------|-------------|
| Gap 1 | Per-folder product toggle round-trip (SNYK_CODE/OSS/IAC/SECRETS_ENABLED) | 4 tests verifying LS→IDE→LS cycle preserves all toggle values |
| Gap 4 | Legacy settings path propagation (SnykProjectSettingsConfigurable) | 4 tests verifying Swing UI writes propagate folder-scope changes correctly |
| Gap 4b | Machine-scope key round-trip (API_ENDPOINT, CLI_PATH, ORGANIZATION, etc.) | 7 tests verifying all 7 round-trip keys + 4 write-only keys survive full cycle |
| Gap 5 | First-folder-to-global-state mapping edge cases | 6 tests covering zero/single/multiple folder configs, addAll edge cases |
| Gap 5b | Coverage gate (80%+ Kover line coverage) | 67 tests raising coverage: SnykApplicationSettingsStateService.kt to 100%, SnykLanguageClient.kt to 86.6% |
| Gap 6 | Organization auto-detection (autoDeterminedOrg, orgSetByUser) | 3 tests verifying end-to-end org auto-detect through folder configs |

**Coverage results after this PR:**

| File | Coverage |
|------|----------|
| SnykApplicationSettingsStateService.kt | 100% |
| SnykLanguageClient.kt | 86.6% |
| FolderConfigSettings.kt | 74.1% (93.5% of unit-testable code; remainder is IntelliJ platform UI) |
| LanguageServerWrapper.kt | 65.9% (untestable lines are LSP process lifecycle code) |

#### Key verifications

- Both UI paths (legacy Swing dialog + HTML/JCEF) produce consistent `LspConfigurationParam` payloads
- Per-folder product toggles survive complete LS→IDE→LS round-trip without data loss
- All 7 machine-scope round-trip keys survive full write-read cycle
- Organization auto-detection (`autoDeterminedOrg`, `orgSetByUser`) works end-to-end
- First-folder-to-global mapping handles zero, one, and many folder configs correctly (ADR-1)
- No production code changes beyond the import fix — all gaps were already handled, just needed test verification

### Checklist

- [x] Read and understood the [Code of Conduct](https://github.com/snyk/snyk-intellij-plugin/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/snyk/snyk-intellij-plugin/blob/master/CONTRIBUTING.md).
- [x] Tests added and all succeed (515 tests, 0 failures)
- [x] Linted (spotlessCheck + ktlintCheck pass)
- [ ] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_No UI changes in this PR — test-only additions._

___

### **PR Type**
Enhancement, Refactoring

___

### **Description**
- Update LSP protocol to v25 and refactor config management.
    - Implement explicit change tracking for LSP settings.
    - Synchronize global and per-folder settings with Language Server.
    - Enhance UI for new configuration model.

___

### Diagram Walkthrough

```mermaid
    flowchart LR
      UI[UI Components] --> SettingsService[SnykApplicationSettingsStateService]
      SettingsService --> LspConfigMapper[LSP Config Generation]
      UI --> FolderConfigurable[SnykProjectSettingsConfigurable]
      FolderConfigurable --> FolderConfigSettings[FolderConfigSettings]
      LspConfigMapper --> LanguageServerWrapper[LanguageServerWrapper]
      LanguageServerWrapper --> SnykLanguageClient[SnykLanguageClient]
      SnykLanguageClient --> LspServer[Language Server]
      SnykLanguageClient --> SettingsService
      SnykLanguageClient --> FolderConfigSettings
      LspServer --> SnykLanguageClient
      SnykLanguageClient -- $/snyk.configuration --> SettingsService
      SnykLanguageClient -- $/snyk.configuration --> FolderConfigSettings
      SettingsService -- Explicit Change Tracking --> LspConfigMapper
      FolderConfigSettings -- Per-Folder Settings --> LspConfigMapper
  ```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Dependencies</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>build.gradle.kts</strong><dd><code>Add Jackson BOM dependency</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-c0dfa6bc7a8685217f70a860145fbdf416d449eaff052fa28352c5cec1a98c06">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td><strong>SnykApplicationSettingsStateService.kt</strong><dd><code>Enhance LSP settings, add explicit change tracking</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-c27dda896e71ad44fa605baf518516de295664a15ee99dbc46af68f5e2303804">+186/-1</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>SnykSettingsDialog.kt</strong><dd><code>Update UI for new LSP settings model and change tracking</code>&nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-09c72a2b8d445fa81616eedf10f2fbb79469dad1a7ff9afff60c29aadeacb634">+17/-7</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SnykTreeSeverityFilterActions.kt</strong><dd><code>Project-specific severity enablement for severity actions</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-00d3745f6fece9164ca2f83f66e16e9cd2eff4767fe2e4bb75e30fe1125bfd22">+22/-4</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SaveConfigHandler.kt</strong><dd><code>Synchronize LSP settings and implement explicit change tracking</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-7a45bcdcd7158b2995c550f9e10be6c2250c2adceee34b2a2f7fc6c591f843f5">+484/-59</a></td>

</tr>

<tr>
  <td><strong>SnykToolWindowSnykScanListener.kt</strong><dd><code>Apply per-file severity filtering using new LSP context</code>&nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-d9acf092f6dc1f381b2d73745cbfc62652be063d88df06898aa289020a93512e">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>AnnotatorCommon.kt</strong><dd><code>Add file-specific severity check and annotation refresh listener</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-d40361829514475bf4d3fb414aa9df13073f358901369997603dc853f2102668">+15/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>LanguageServerWrapper.kt</strong><dd><code>Refactor LSP settings payload generation with explicit change tracking</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-7ab8b7f2e05d19f7d673c7978b0e9108af49a56d83c6c854676dac76eb4ec377">+241/-40</a></td>

</tr>

<tr>
  <td><strong>SnykLanguageClient.kt</strong><dd><code>Update LSP config handler, sync folder config to global state</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-3c1953a8d970fec662c910f9816060c1441599f1c4fc14e51de17a616825b55b">+225/-16</a></td>

</tr>
</table></details></td></tr><tr><td><strong>Refactoring</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td><strong>SnykProjectSettingsConfigurable.kt</strong><dd><code>Update folder config apply logic for LSP settings</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-fba6e4ba36040991ed55cb219d676f29e4de77fd2b2eb6fd88f1efae12cf27a4">+14/-6</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ReferenceChooserDialog.kt</strong><dd><code>Update reference chooser for LSP settings structure</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-42ce6b3e9ae635b7afeb73139e194253e330c8bf984e4c8cc8680fbb4be24ac9">+22/-12</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>ConfigurationDataClasses.kt</strong><dd><code>Update LSP configuration data classes and naming conventions</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-f6182041b276430b8e0472fd1d76eb89347eefaeec76fb004fda0ae2d2e3a215">+75/-36</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>SnykToolWindowPanel.kt</strong><dd><code>Update tool window for new LSP settings keys</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-602f985789e7372169095fb4e8c819e382fc1ef9358346c12ca0fa4a0c14720c">+10/-5</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SnykAnnotator.kt</strong><dd><code>Use file-specific severity filtering in annotator</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-0e5198fcc8ea092d80c71fc6acbbadd84aa5452a9192c5c1eb68c27f34a34127">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Types.kt</strong><dd><code>Update LSP types and add helper functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-b95caa751133ff762c10e6bc88f4e5c332662e95554ae400edd89222aaa93e9c">+0/-32</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>FolderConfigSettings.kt</strong><dd><code>Adapt FolderConfigSettings to LspFolderConfig and new settings model</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-ae81e594ea55c609651ba3b90e8ca700a9015648917ea57322eca1352403592a">+152/-63</a></td>

</tr>

<tr>
  <td><strong>LanguageServerSettings.kt</strong><dd><code>Deprecate LanguageServerSettings, introduce LspConfigurationParam, </code><br><code>ConfigSetting</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-32b6d1fb82acd4563888eed32e8252119fca29a67ba4da4c35ca3693e79f6d1c">+52/-61</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td><strong>LsFolderSettingsKeys.kt</strong><dd><code>Define new LSP settings keys for folder-specific configurations</code></dd></td>
  <td><a href=""></a></td>

</tr>

<tr>
  <td><strong>LsSettingsKeys.kt</strong><dd><code>Define new LSP settings keys for machine-specific configurations</code></dd></td>
  <td><a href=""></a></td>

</tr>

<tr>
  <td><strong>plugin.xml</strong><dd><code>Enable new config dialog by default and update registry keys</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-13198d14dd3abf13abd9cb906331ed7d9663fc081757b6845fd5f3c1850b5693">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>14 files</summary><table>
<tr>
  <td><strong>SnykApplicationSettingsStateServiceTest.kt</strong><dd><code>Add tests for new LSP settings and explicit change tracking</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-62b5591535de5417707b8a79875eaa2b6dbe8055205947b0d5950e0739457142">+343/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>SnykCliAuthenticationServiceTest.kt</strong><dd><code>Clean up mock setup in test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-9203fbd79dac895ba5d22ecbc4b3f66c238b47c886ff3deee696c31b786be808">+11/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SettingsApplyFunctionsTest.kt</strong><dd><code>Add tests for folder-level change tracking and parsing logic</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-600b036cee5fb5efed66ca8f274a888688ab9145c146e7d89ebae2a18a700534">+33/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SnykProjectSettingsConfigurableTest.kt</strong><dd><code>Update tests for LSP folder config changes and explicit flags</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-fb13d31b0bf5f7e98ef183d518c0abbcbf376bf96940655328458c5d5984db39">+429/-15</a></td>

</tr>

<tr>
  <td><strong>SaveConfigHandlerTest.kt</strong><dd><code>Update tests for new LSP settings model and change tracking logic</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-622128219f9ae8c6bb3755bc900be83782411e0cd27b94762989ea994f6197cd">+991/-29</a></td>

</tr>

<tr>
  <td><strong>SnykAuthPanelIntegTest.kt</strong><dd><code>Clean up mock setup in test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-ef354fe5e1a5278b0c47c3bef5b5aa19fe1251f99bb6bde9cbde02096460902f">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SnykToolWindowSnykScanListenerTest.kt</strong><dd><code>Update tests for new LSP severity filtering and folder config </code><br><code>structure</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-63fcaaef88f2d63eff511b2771ae72cd591264da4d91230167879460e1fcd97d">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>AnnotatorCommonIntegTest.kt</strong><dd><code>Test file-specific severity filtering and annotation refresh logic</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-57fe620156b92d5bdbf17b89646e8ee2b8716e7ccaccaeb2bd3e10d2d361d1de">+100/-4</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>AnnotatorCommonTest.kt</strong><dd><code>Test AnnotatorCommon initialization and severity checks</code>&nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-93d12df81bb156b6d1a363c1a16cf257448904793cadc7f7dca821a2433520af">+81/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>LanguageServerWrapperTest.kt</strong><dd><code>Add tests for new LSP settings payload generation and change tracking</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-00f3f6aa0e06837878a80b44f86354478d04c5cad631b47caa1298826bd33ce3">+436/-12</a></td>

</tr>

<tr>
  <td><strong>SnykLanguageClientTest.kt</strong><dd><code>Update tests for new LSP config handling and explicit change tracking</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-280846bfaa62f843b7da34110e6fcb4ff40d48f243505f6bc887b4d46a32ac84">+1280/-8</a></td>

</tr>

<tr>
  <td><strong>FolderConfigSettingsTest.kt</strong><dd><code>Test new LspFolderConfig structure and path normalization logic</code></dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-e10c2bd2bd3d653016af46ba06cae74768aae3aaa6cfef711481236e530b77cc">+604/-106</a></td>

</tr>

<tr>
  <td><strong>TestFolderConfigHelper.kt</strong><dd><code>Helper function for creating LspFolderConfig instances</code>&nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-e0c70b01b98686e313e261c66ba4ebb125eadb29f0bea6e272557cb8824cd449">+37/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>WorkspaceTrustServiceIntegrationTest.kt</strong><dd><code>Clean up mock setup in test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-d0f4f42f8227618b87459a1d79b2c3d6ebea3043f2318d814cec27c598d6d6d0">+13/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Miscellaneous</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>settings.json</strong><dd><code>Remove unused VS Code settings file</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-a5de3e5871ffcc383a2294845bd3df25d3eeff6c29ad46e3a396577c413bf357">+0/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Additional files</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>LanguageServerSettingsKeys.kt</strong></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-cc228896b883b94ac37a5157263a5629f4f1e90c68db05c75d9b97004850eefc">+69/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ReferenceChooserDialogTest.kt</strong></td>
  <td><a href="https://github.com/snyk/snyk-intellij-plugin/pull/797/changes#diff-7b066c1d6c0ea32ee7f96d27124f1b6c58387683454f72205776b0b82e7514ae">+32/-16</a>&nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: src/main/kotlin/io/snyk/plugin/services/SnykApplicationSettingsStateService.kt [hunk 1]
```diff
@@ -40,6 +40,24 @@ class SnykApplicationSettingsStateService :
   // testing flag
   var fileListenerEnabled: Boolean = true
 
+  var explicitChanges: MutableSet<String> = mutableSetOf()
+
+  // folder path -> set of setting keys explicitly changed for that folder
+  var folderExplicitChanges: MutableMap<String, MutableSet<String>> = mutableMapOf()
+
+  fun markExplicitlyChanged(settingKey: String) {
```
Comment (nick-y-snyk) [inline src/main/kotlin/io/snyk/plugin/services/SnykApplicationSettingsStateService.kt:68]: what about removal? Reset flows:
- from html settings page
- field removal from the settings file

File: src/main/kotlin/io/snyk/plugin/ui/jcef/ConfigurationDataClasses.kt [hunk 2]
```diff
@@ -113,6 +113,16 @@ data class FolderConfigData(
   @SerializedName("orgSetByUser") val orgSetByUser: Boolean? = null,
   @SerializedName("scanCommandConfig")
   val scanCommandConfig: Map<String, ScanCommandConfigData>? = null,
+  // Org-scope override fields
+  @SerializedName("scanAutomatic") val scanAutomatic: Boolean? = null,
```
Comment (nick-y-snyk) [inline src/main/kotlin/io/snyk/plugin/ui/jcef/ConfigurationDataClasses.kt]: During vscode pr creation, I discovered that data structure coming from html form is not using ls key names, which complicates each IDE implementation. 

I fixed that here - https://github.com/snyk/snyk-ls/pull/1196 (snakecasing and changing field names to match). 
In vscode code PR, this change + "settings registry" implementation removed over 1k lines of duplicated mappers and boilerplate, since now there are only ls keys and ide persistence keys

File: src/main/kotlin/io/snyk/plugin/ui/jcef/SaveConfigHandler.kt [hunk 3]
```diff
@@ -191,6 +194,8 @@ class SaveConfigHandler(
     if (!isFallback) {
       config.folderConfigs?.let { applyFolderConfigs(it) }
     }
+
+    LanguageServerWrapper.getInstance(project).updateConfiguration()
```
Comment (nick-y-snyk) [inline src/main/kotlin/io/snyk/plugin/ui/jcef/SaveConfigHandler.kt]: how did it work before without this notification?

File: src/main/kotlin/io/snyk/plugin/ui/jcef/SaveConfigHandler.kt [hunk 4]
```diff
@@ -199,28 +204,56 @@ class SaveConfigHandler(
...
+    config.manageBinariesAutomatically?.let {
+      settings.manageBinariesAutomatically = it
+      settings.markExplicitlyChanged(LsSettingsKeys.AUTOMATIC_DOWNLOAD)
+    }
 
     // Use the provided cliPath from the config if present, or the default CLI path if not.
-    config.cliPath?.let { path -> settings.cliPath = path.ifEmpty { getDefaultCliPath() } }
+    config.cliPath?.let { path ->
+      settings.cliPath = path.ifEmpty { getDefaultCliPath() }
+      settings.markExplicitlyChanged(LsSettingsKeys.CLI_PATH)
```
Comment (nick-y-snyk) [inline src/main/kotlin/io/snyk/plugin/ui/jcef/SaveConfigHandler.kt]: it needs to be diff based, looks like it will always send changed: true

File: src/main/kotlin/io/snyk/plugin/ui/jcef/ConfigurationDataClasses.kt (hunk 2 above)
Comment (nick-y-snyk) [reply src/main/kotlin/io/snyk/plugin/ui/jcef/ConfigurationDataClasses.kt]: vscode pr - https://github.com/snyk/vscode-extension/pull/733

File: CLAUDE.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
+# Project Rules
```
Comment (nick-y-snyk) [inline CLAUDE.md]: Please don't include it and .claude

File: tests.json [hunk 6]
```diff
@@ -0,0 +1,51 @@
+{
```
Comment (nick-y-snyk) [inline tests.json]: local file

File: test-output.txt [hunk 7]
```diff
@@ -0,0 +1,10 @@
+> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
```
Comment (nick-y-snyk) [inline test-output.txt]: local file

File: src/main/kotlin/snyk/common/lsp/settings/FolderConfigSettings.kt [hunk 8]
```diff
@@ -144,7 +156,12 @@ class FolderConfigSettings {
    */
   fun setAutoOrganization(project: Project, autoOrganization: Boolean) {
     getFolderConfigs(project).forEach { folderConfig ->
-      val updatedConfig = folderConfig.copy(orgSetByUser = !autoOrganization)
+      val updatedConfig =
+        folderConfig.withSetting(
+          LsFolderSettingsKeys.ORG_SET_BY_USER,
+          !autoOrganization,
+          changed = true,
```
Comment (nick-y-snyk) [inline src/main/kotlin/snyk/common/lsp/settings/FolderConfigSettings.kt:167]: can we omit this for folderConfig fields, since user overrides are processed on LS side? also applies to other places

**Review by nick-y-snyk — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/snyk/plugin/ui/jcef/SaveConfigHandler.kt (hunk 3 above)
Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/ui/jcef/SaveConfigHandler.kt]: **Re: SaveConfigHandler.kt:171** — Before this change, settings were applied to `SnykApplicationSettingsStateService` but the Language Server was not explicitly notified. The LS would pick up changes either on the next periodic sync or when it needed to read settings. Adding `updateConfiguration()` ensures the LS receives the updated configuration immediately after the user saves, rather than relying on the next sync cycle. This is important for settings like product toggles or severity filters where the user expects immediate effect.

**Re: ConfigurationDataClasses.kt:117 (LS key name alignment)** — Great catch. Once snyk-ls#1196 lands and the HTML form uses LS key names directly, the IntelliJ plugin can adopt the same approach and remove the mapping layer in `SaveConfigHandler`. We'll track this as a follow-up after the LS change is merged.

**Re: remaining comments** — File removal (CLAUDE.md, tests.json, test-output.txt), diff-based change tracking, removal of folder-config markExplicitlyChanged calls, and reset/removal flows are all being addressed in follow-up commits on this branch.

**Review by nick-y-snyk — APPROVED** [review]
